src/hotspot/share/gc/g1/g1MMUTracker.hpp

```diff
@@ -77,25 +39,43 @@ class G1MMUTrackerQueueElem {
   inline double end_time()   { return _end_time; }
   inline double duration()   { return _end_time - _start_time; }
 
-  G1MMUTrackerQueueElem() {
+  G1MMUTrackerElem() {
     _start_time = 0.0;
     _end_time   = 0.0;
   }
 
-  G1MMUTrackerQueueElem(double start_time, double end_time) {
+  G1MMUTrackerElem(double start_time, double end_time) {
     _start_time = start_time;
     _end_time   = end_time;
   }
 };
 
-// this is an implementation of the MMUTracker using a (fixed-size) queue
-// that keeps track of all the recent pause times
-class G1MMUTrackerQueue: public G1MMUTracker {
+
+// This implementation uses a fixed-size queue to keep track of all
+// recent pause times. Two major user controls over G1 behavior are
```

Comment:
I would move the first sentence down to right before the warning about seconds. I would also change it to something like:
`The G1MMUTracker uses a fixed-size queue to keep track of all recent pause times. The pause time data is used to avoid breaking the MMU.`

Reply:
done